diff --git a/backends/gf-gpu-xrandr.c b/backends/gf-gpu-xrandr.c
index f0e8c62..9be7454 100644
--- a/backends/gf-gpu-xrandr.c
+++ b/backends/gf-gpu-xrandr.c
@@ -145,7 +145,7 @@ gf_gpu_xrandr_read_current (GfGpu   *gpu,
   monitor_manager->screen_width = WidthOfScreen (screen);
   monitor_manager->screen_height = HeightOfScreen (screen);
 
-  resources = XRRGetScreenResourcesCurrent (xdisplay, DefaultRootWindow (xdisplay));
+  resources = XRRGetScreenResources (xdisplay, DefaultRootWindow (xdisplay));
 
   if (!resources)
     {
